lib/posix-process/signal/rt_sigtimedwait.c

```diff
@@ -64,9 +69,30 @@ UK_LLSYSCALL_R_DEFINE(int, rt_sigtimedwait,
 		      const struct timespec *, timeout,
 		      size_t, sigsetsize)
 {
-	UK_WARN_STUBBED();
+	struct uk_thread *current = uk_thread_current();
+	__nsec timeout_ns;
+	int rc;
+
+	if (timeout) {
+		if (unlikely(!uk_time_spec_canonical(timeout))
+			return -EINVAL;
+		timeout_ns = ukplat_monotonic_clock() +
+			ukarch_time_sec_to_nsec(timeout->tv_sec) +
+			timeout->tv_nsec;
+		uk_thread_block_until(current, (__snsec) timeout_ns);
```

Comment:
nitpick but no space after cast (i know that's just how it was initially when you picked it up)? it;s just a checkpatch `CHECK` but as you know (and as you also do) we tend to respect that :sweat_smile: